src/descriptor/mod.rs

```diff
@@ -202,23 +203,31 @@ impl DescriptorInfo {
     /// should use the method [DescriptorPublicKey::parse_descriptor] to obtain the
     /// Descriptor and a secret key to public key mapping
     pub fn from_desc_str(s: &str) -> Result<Self, Error> {
-        let is_secret_key = |s: &String, has_secret: &mut bool| -> String {
-            *has_secret = match DescriptorSecretKey::from_str(s) {
-                Ok(_sk) => true,
-                Err(_) => false,
-            };
-            String::from("")
-        };
+        struct HasSecrets(bool);
+
+        impl Translator<String, String, Error> for HasSecrets {
+            fn pk(&mut self, pk: &String) -> Result<String, Error> {
+                match DescriptorSecretKey::from_str(pk) {
+                    Ok(_sk) => true,
+                    Err(_) => false,
+                };
+                Ok(String::from(""))
```

Comment:
This was a bug! I think I should really add Clippy to this to help in rebase. During updating the translator API, I need to store the value in the translator struct which I forgot to write.

Reply:
This method also did not exist upstream, which is why it did not have any tests that failed.

Reply:
Nice! Ok, this makes sense now.

Though I wonder if the target keytype should be `()` rather than `String` now.

Reply:
I guess `()` is not a `MiniscriptKey` ... but `DummyKey` is. Maybe we should rename that to `NullKey` or something so it could be the "standard way" to handle cases like this?

Or maybe using `Translator` is wrong and is what the `for_each` API is for?

Reply:
Yep, I think what I need here is `ForEach` instead of Translator. I used the empty String to avoid allocation of the entire string again because the subsequent code only tries to infer the descriptor type by looking at the first few characters of the descriptor string and does not care about the entire string.